### PR TITLE
fix: Allow CLI version command to succeed without server connection (#3049)

### LINKED ISSUE #3049: argocd version command is not working on 1.4.x
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Can't use cli `version` command

**To Reproduce**

```shell
docker run --rm argoproj/argocd:v1.4.2 argocd version
time="2020-01-29T15:09:10Z" level=fatal msg="Argo CD server address unspecified"

docker run --rm argoproj/argocd:v1.4.2 argocd --help 
argocd controls a Argo CD server

Usage:
  argocd [flags]
  argocd [command]

Available Commands:
  account     Manage account settings
  app         Manage applications
  cert        Manage repository certificates and SSH known hosts entries
  cluster     Manage cluster credentials
  completion  output shell completion code for the specified shell (bash or zsh)
  context     Switch between contexts
  help        Help about any command
  login       Log in to Argo CD
  logout      Log out from Argo CD
  proj        Manage projects
  relogin     Refresh an expired authenticate token
  repo        Manage repository connection parameters
  repocreds   Manage repository connection parameters
  version     Print version information

Flags:
      --auth-token string               Authentication token
      --config string                   Path to Argo CD config (default "/home/argocd/.argocd/config")
      --grpc-web                        Enables gRPC-web protocol. Useful if Argo CD server is behind proxy which does not support HTTP2.
  -H, --header strings                  Sets additional header to all requests made by Argo CD CLI. (Can be repeated multiple times to add multiple headers, also supports comma separated headers)
  -h, --help                            help for argocd
      --insecure                        Skip server certificate and domain verification
      --loglevel string                 Set the logging level. One of: debug|info|warn|error (default "info")
      --plaintext                       Disable TLS
      --port-forward                    Connect to a random argocd-server port using port forwarding
      --port-forward-namespace string   Namespace name which should be used for port forwarding
      --server string                   Argo CD server address
      --server-crt string               Server certificate file

Use "argocd [command] --help" for more information about a command.
```

**Expected behavior**

```
docker run --rm argoproj/argocd:v1.3.6 argocd version 
argocd: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:46:45Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
time="2020-01-29T15:19:30Z" level=fatal msg="Argo CD server address unspecified"
```

**Version**

eeeer :smile: 

spotted while rising #3048

### PR DESCRIPTION
**Pull request by shuheiktgw**
Fixes https://github.com/argoproj/argo-cd/issues/3049.

- Fixed version command to print client version without server.
- Fixed version command `-o yaml | json --client` to work without server.

Thank you for your review!

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: cmd/argocd/commands/version.go [hunk 1]
```diff
@@ -86,38 +81,53 @@ func NewVersionCmd(clientOpts *argocdclient.ClientOptions) *cobra.Command {
...
-		fmt.Printf("  GoVersion: %s\n", version.GoVersion)
-		fmt.Printf("  Compiler: %s\n", version.Compiler)
-		fmt.Printf("  Platform: %s\n", version.Platform)
+
+	if short {
+		return
 	}
-	if client {
+
+	fmt.Printf("%s: %s\n", cliName, version)
```
Comment (jannfis) [inline cmd/argocd/commands/version.go:101]: I think this one prints the client version a second time, given that it is also printed just after entering `printClientVersion` method.

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks a lot for your PR @shuheiktgw! It looks good to me so far, I have tested the changes locally and they work pretty well!

I have just found one minor quirk, addressed below. Please review & fix if appropriate!

File: cmd/argocd/commands/version.go (hunk 1 above)
Comment (shuheiktgw) [reply cmd/argocd/commands/version.go]: @jannfis Thank you so much for taking your time and reviewing the PR. I've just fixed it and rebased the PR, so please confirm the change! 👍

**Review by jannfis — APPROVED** [review]
LGTM. thank you very much @shuheiktgw !

**Pull request merged**